refactor: Use `pathlib` for manipulating paths

romanc opened the pull request:
**Description**

Use `pathlib.Path` objects for "eta files" in tests. This is in preparation of NDSL moving towards `pathlib` objects and enforcing them for eta files, see https://github.com/NOAA-GFDL/NDSL/pull/149.

**How Has This Been Tested?**

Running affected tests locally.

**Checklist:**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas: N/A
- [ ] I have made corresponding changes to the documentation: N/A
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules: N/A
- [ ] New check tests, if applicable, are included: N/A

romanc committed:
tests: Use pathlib.Path for eta_files

romanc commented:
@fmalatino @oelbert please review

/cc @FlorianDeconinck 

(tagging people because I don't have rights to add reviewers)

FlorianDeconinck reviewed (dismissed):
LGTM

romanc commented:
(Feel free to merge whenever you guys are happy. I don't have rights to push changes in this repo.)

FlorianDeconinck commented:
I know Frank try to give us write access but I don't seem to have them either

fmalatino commented:
> I know Frank try to give us write access but I don't seem to have them either

@FlorianDeconinck  and @romanc, you both have write access, but would need `maintain` access to have merge power, which i don't think I have the ability to grant. I will let one more review from our side come in and then this will be merged.

FlorianDeconinck commented:
Makes sense!

romanc commented:
All good, Frank. I'm already happy that I have write access and can add reviewers through the interface. Here I'm just making sure PRs move forward. I can be a bit impatient at times when I work on many smaller PRs...

fmalatino commented:
> All good, Frank. I'm already happy that I have write access and can add reviewers through the interface. Here I'm just making sure PRs move forward. I can be a bit impatient at times when I work on many smaller PRs...

@romanc , right there with you. I will do my best to make sure nothing gets held up too long.

romanc commented:
Friendly reminder @oelbert

oelbert commented:
"friendly" lol

oelbert reviewed (commented):
I think there are some paths to find in `test_restart_fortran.py`, also do we need to also update typehints for the grid configs in `pace/grid.py`?

oelbert reviewed (changes requested):
Meant to request changes, sorry

oelbert commented:
Also, it would be nice to update the ipython notebooks as well

romanc commented:
@oelbert sounds all doable. I'll take a stab at it on Monday :broom:

oelbert commented:
Great, thanks Roman!

romanc committed:
Udpate more paths from str to Path

romanc committed:
Revert changes to pace/comm.py and pace/configs/comm.py

romanc commented:
@oelbert I pushed some more updates from `str` to `Path`

- Search for `os.path` and replaced the verbose that path handling with `pathlib` paths.
- Added `tests/paths.py` to centralize some paths like `EXAMPLE_CONFIGS_DIR` in tests.

> Do we need to also update typehints for the grid configs in `pace/grid.py`?

Currently, no. That wouldn't work since NDSL (without https://github.com/NOAA-GFDL/NDSL/pull/149 merged) expects things to be strings. We can change this in a follow-up PR.

> Also, it would be nice to update the ipython notebooks as well

Can you point me to what you'd like to update? At a first glance, I didn't find path handling in the example notebooks. Or maybe this could be a follow-up PR? The notebooks look like they could use some love from someone with domain knowledge ...

romanc commented:
One question that popped up (maybe unrelated or not in scope for this PR) is: What's the difference between `pace/comm.py` and `pace/config/comm.py`? They are almost identical. Tests complained about changes in `pace/comm.py` but not about the same changes in `pace/config/comm.py`. That seems suspicious and makes me think that maybe `pace/config/comm.py` is unused? I guess someone with history knowledge should weigh in here...

oelbert reviewed (dismissed):
I can separately update the notebooks if needed after this goes in. Nice job!

oelbert commented:
> One question that popped up (maybe unrelated or not in scope for this PR) is: What's the difference between `pace/comm.py` and `pace/config/comm.py`? They are almost identical. Tests complained about changes in `pace/comm.py` but not about the same changes in `pace/config/comm.py`. That seems suspicious and makes me think that maybe `pace/config/comm.py` is unused? I guess someone with history knowledge should weigh in here...

Looking at the code, it seems like somehow the files got duplicated and we kept developing `pace/comm.py` and `pace/configs/comm.py` can be deleted, but that's a different PR I'd say

romanc commented:
> > One question that popped up (maybe unrelated or not in scope for this PR) is: What's the difference between `pace/comm.py` and `pace/config/comm.py`? They are almost identical. Tests complained about changes in `pace/comm.py` but not about the same changes in `pace/config/comm.py`. That seems suspicious and makes me think that maybe `pace/config/comm.py` is unused? I guess someone with history knowledge should weigh in here...
> 
> Looking at the code, it seems like somehow the files got duplicated and we kept developing `pace/comm.py` and `pace/configs/comm.py` can be deleted, but that's a different PR I'd say

Okay. In that case, let me undo all changes to `pace/configs/comm.py` in this PR (no need to change anything that's about to be deleted) and open a separate PR proposing the deletion of `pace/configs/comm.py`.
_Edit_ PR is here: https://github.com/NOAA-GFDL/pace/pull/126

> I can separately update the notebooks if needed after this goes in. Nice job!

Thanks. I'd appreciate you looking to the notebooks in a follow-up PR.

fmalatino commented:
Is this good to go?

romanc commented:
> Is this good to go?

I'm happy, but let's Oliver have the last word here.

oelbert reviewed (approved):
Sorry, thought I'd already approved this PR
